io_uring test failure

### Zig Version

0.9.0-dev.1812+1ea650bb7

### Steps to Reproduce

```
$ uname -a 
Linux ark 5.4.122 #1-NixOS SMP Wed May 26 10:05:21 UTC 2021 x86_64 GNU/Linux
$ ./zig test ../lib/std/std.zig
```


### Expected Behavior

All std lib tests passed.

### Actual Behavior

After #10151 was merged, this started happening:

```
Test [293/2086] os.linux.io_uring.test "timeout_link_chain1"... Req 0x22222222 got -9
Test [294/2086] os.linux.io_uring.test "timeout_link_chain1"... FAIL (TestUnexpectedResult)
/home/andy/Downloads/zig/lib/std/io/writer.zig:23:26: 0x5d84b1 in io.writer.Writer(*io.fixed_buffer_stream.FixedBufferStream([]u8),io.fixed_buffer_stream.WriteError,io.fixed_buffer_stream.FixedBufferStream([]u8).write).writeAll (test)
                index += try self.write(bytes[index..]);
                         ^
/home/andy/Downloads/zig/lib/std/io/fixed_buffer_stream.zig:63:46: 0x5d801a in io.fixed_buffer_stream.FixedBufferStream([]u8).write (test)
            if (self.pos >= self.buffer.len) return error.NoSpaceLeft;
                                             ^
/home/andy/Downloads/zig/lib/std/io/writer.zig:17:13: 0x5d85a2 in io.writer.Writer(*io.fixed_buffer_stream.FixedBufferStream([]u8),io.fixed_buffer_stream.WriteError,io.fixed_buffer_stream.FixedBufferStream([]u8).write).write (test)
            return writeFn(self.context, bytes);
            ^
/home/andy/Downloads/zig/lib/std/io/writer.zig:23:26: 0x5d84b1 in io.writer.Writer(*io.fixed_buffer_stream.FixedBufferStream([]u8),io.fixed_buffer_stream.WriteError,io.fixed_buffer_stream.FixedBufferStream([]u8).write).writeAll (test)
                index += try self.write(bytes[index..]);
                         ^
/home/andy/Downloads/zig/lib/std/io/fixed_buffer_stream.zig:63:46: 0x5d801a in io.fixed_buffer_stream.FixedBufferStream([]u8).write (test)
            if (self.pos >= self.buffer.len) return error.NoSpaceLeft;
                                             ^
/home/andy/Downloads/zig/lib/std/io/writer.zig:17:13: 0x5d85a2 in io.writer.Writer(*io.fixed_buffer_stream.FixedBufferStream([]u8),io.fixed_buffer_stream.WriteError,io.fixed_buffer_stream.FixedBufferStream([]u8).write).write (test)
            return writeFn(self.context, bytes);
            ^
/home/andy/Downloads/zig/lib/std/io/writer.zig:23:26: 0x5d84b1 in io.writer.Writer(*io.fixed_buffer_stream.FixedBufferStream([]u8),io.fixed_buffer_stream.WriteError,io.fixed_buffer_stream.FixedBufferStream([]u8).write).writeAll (test)
                index += try self.write(bytes[index..]);
                         ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:39:27: 0x77ffaf in os.linux.io_uring.IO_Uring.init_params (test)
        if (entries == 0) return error.EntriesZero;
                          ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:31:16: 0x663f3d in os.linux.io_uring.IO_Uring.init (test)
        return try IO_Uring.init_params(entries, &params);
               ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:40:46: 0x77fff6 in os.linux.io_uring.IO_Uring.init_params (test)
        if (!std.math.isPowerOfTwo(entries)) return error.EntriesNotPowerOfTwo;
                                             ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:31:16: 0x663f3d in os.linux.io_uring.IO_Uring.init (test)
        return try IO_Uring.init_params(entries, &params);
               ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:143:46: 0x7818e9 in os.linux.io_uring.IO_Uring.get_sqe (test)
        if (next -% head > self.sq.sqes.len) return error.SubmissionQueueFull;
                                             ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:354:21: 0x664228 in os.linux.io_uring.IO_Uring.nop (test)
        const sqe = try self.get_sqe();
                    ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:1504:36: 0x46e7ae in os.linux.io_uring.test "write/read" (test)
    if (cqe_write.err() == .INVAL) return error.SkipZigTest;
                                   ^
/home/andy/Downloads/zig/lib/std/os.zig:2127:19: 0x9cfe9f in os.unlinkatZ (test)
        .NOENT => return error.FileNotFound,
                  ^
/home/andy/Downloads/zig/lib/std/fs.zig:1633:25: 0x782f9b in fs.Dir.deleteFileZ (test)
            else => |e| return e,
                        ^
/home/andy/Downloads/zig/lib/std/fs.zig:1612:13: 0x6654c7 in fs.Dir.deleteFile (test)
            return self.deleteFileZ(&sub_path_c);
            ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:1616:9: 0x470312 in os.linux.io_uring.test "openat" (test)
        return error.SkipZigTest;
        ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:1646:36: 0x470b05 in os.linux.io_uring.test "close" (test)
    if (cqe_close.err() == .INVAL) return error.SkipZigTest;
                                   ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:1686:37: 0x4714b6 in os.linux.io_uring.test "accept/connect/send/recv" (test)
    if (cqe_accept.err() == .INVAL) return error.SkipZigTest;
                                    ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:1827:9: 0x472e6e in os.linux.io_uring.test "timeout_remove" (test)
        return error.SkipZigTest;
        ^
/home/andy/Downloads/zig/lib/std/testing.zig:306:14: 0x5c0d5b in testing.expect (test)
    if (!ok) return error.TestUnexpectedResult;
             ^
/home/andy/Downloads/zig/lib/std/os/linux/io_uring.zig:1892:21: 0x473718 in os.linux.io_uring.test "timeout_link_chain1" (test)
                    try testing.expect(false);
                    ^
/home/andy/Downloads/zig/lib/std/unicode.zig:49:46: 0x8c92af in unicode.utf8Encode (test)
            if (0xd800 <= c and c <= 0xdfff) return error.Utf8CannotEncodeSurrogateHalf;
                                             ^
/home/andy/Downloads/zig/lib/std/unicode.zig:13:5: 0x94eb32 in unicode.utf8CodepointSequenceLength (test)
    return error.CodepointTooLarge;
    ^
/home/andy/Downloads/zig/lib/std/unicode.zig:36:20: 0x8c91f2 in unicode.utf8Encode (test)
    const length = try utf8CodepointSequenceLength(c);
                   ^
/home/andy/Downloads/zig/lib/std/fmt.zig:1769:25: 0x5cf22a in fmt.charToDigit (test)
    if (value >= radix) return error.InvalidCharacter;
                        ^
/home/andy/Downloads/zig/lib/std/fmt.zig:2386:20: 0x5c456d in fmt.hexToBytes (test)
        const lo = try charToDigit(input[in_i + 1], 16);
                   ^
/home/andy/Downloads/zig/lib/std/fmt.zig:2379:9: 0x5c446f in fmt.hexToBytes (test)
        return error.InvalidLength;
        ^
/home/andy/Downloads/zig/lib/std/fmt.zig:2381:9: 0x5c44b3 in fmt.hexToBytes (test)
        return error.NoSpaceLeft;
        ^
/home/andy/Downloads/zig/lib/std/io/fixed_buffer_stream.zig:63:46: 0x5d801a in io.fixed_buffer_stream.FixedBufferStream([]u8).write (test)
            if (self.pos >= self.buffer.len) return error.NoSpaceLeft;
                                             ^
/home/andy/Downloads/zig/lib/std/io/writer.zig:17:13: 0x5d85a2 in io.writer.Writer(*io.fixed_buffer_stream.FixedBufferStream([]u8),io.fixed_buffer_stream.WriteError,io.fixed_buffer_stream.FixedBufferStream([]u8).write).write (test)
            return writeFn(self.context, bytes);
            ^
```

The error `Semantic Analysis [3438/4623] Killed` is gone when I increased memory for the virtual machine to 8192MiB with the following command (4096MiB was not enough).

```
lxc stop bionic-vm
lxc config set bionic-vm limits.memory 4096MiB
lxc start bionic-vm
```

I got the following error when I ran `./zig test ./lib/std/std.zig`.

```
./lib/std/compress/gzip.zig:191:20: error: unable to find '/root/zig-linux-x86_64-0.9.0-dev.1815+20e19e75f/lib/std/compress/rfc1952.txt.gz'
        @embedFile("rfc1952.txt.gz"),
                   ^
./lib/std/compress/gzip.zig:191:9: note: referenced here
        @embedFile("rfc1952.txt.gz"),
        ^
./lib/std/compress/zlib.zig:128:20: error: unable to find '/root/zig-linux-x86_64-0.9.0-dev.1815+20e19e75f/lib/std/compress/rfc1951.txt.z.0'
        @embedFile("rfc1951.txt.z.0"),
                   ^
./lib/std/compress/zlib.zig:128:9: note: referenced here
        @embedFile("rfc1951.txt.z.0"),
        ^
./lib/std/crypto/aegis.zig:351:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("79d94593d8c2119d7e8fd9b8fc77845c5c077a05b2528b6ac54b563aed8efe84", &c);
             ^
./lib/std/crypto/aegis.zig:374:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("41de9000a7b5e40e2d68bb64d99ebb19", &c);
             ^
./lib/std/crypto/aegis.zig:391:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("83cc600dc4e3e7e62d4055826174f149", &tag);
             ^
./lib/std/crypto/aegis.zig:407:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("f373079ed84b2709faee373584585d60accd191db310ef5d8b11833df9dec711", &c);
             ^
./lib/std/crypto/aegis.zig:430:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("b98f03a947807713d75a4fff9fc277a6", &c);
             ^
./lib/std/crypto/aegis.zig:447:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("f7a0878f68bd083e8065354071fc27c3", &tag);
             ^
./lib/std/crypto/aes_gcm.zig:113:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("6b6ff610a16fa4cd59f1fb7903154e92", &tag);
             ^
./lib/std/crypto/aes_gcm.zig:125:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("262ed164c2dfb26e080a9d108dd9dd4c", &tag);
             ^
./lib/std/crypto/aes_gcm.zig:141:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("5ca1642d90009fea33d01f78cf6eefaf01d539472f7c", &c);
             ^
./lib/std/crypto/aes_gcm.zig:158:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("5ca1642d90009fea33d01f78cf6eefaf01", &c);
             ^
./lib/std/crypto/blake2.zig:204:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Blake2s160, h1, "");
             ^
./lib/std/crypto/blake2.zig:223:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/blake2.zig:275:18: error: container 'crypto.test' has no member called 'assertEqualHash'
        try htest.assertEqualHash(Blake2s160, h1, block[0..]);
                 ^
./lib/std/crypto/blake2.zig:287:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Blake2s224, h1, "");
             ^
./lib/std/crypto/blake2.zig:306:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/blake2.zig:357:18: error: container 'crypto.test' has no member called 'assertEqualHash'
        try htest.assertEqualHash(Blake2s224, h1, block[0..]);
                 ^
./lib/std/crypto/blake2.zig:369:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Blake2s256, h1, "");
             ^
./lib/std/crypto/blake2.zig:388:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/blake2.zig:425:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/blake2.zig:449:18: error: container 'crypto.test' has no member called 'assertEqualHash'
        try htest.assertEqualHash(Blake2s256, h1, block[0..]);
                 ^
./lib/std/crypto/blake2.zig:627:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Blake2b160, h1, "");
             ^
./lib/std/crypto/blake2.zig:646:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/blake2.zig:704:18: error: container 'crypto.test' has no member called 'assertEqualHash'
        try htest.assertEqualHash(Blake2b160, h1, block[0..]);
                 ^
./lib/std/crypto/blake2.zig:716:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Blake2b384, h1, "");
             ^
./lib/std/crypto/blake2.zig:735:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/blake2.zig:793:18: error: container 'crypto.test' has no member called 'assertEqualHash'
        try htest.assertEqualHash(Blake2b384, h1, block[0..]);
                 ^
./lib/std/crypto/blake2.zig:805:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Blake2b512, h1, "");
             ^
./lib/std/crypto/blake2.zig:824:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/blake2.zig:861:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/blake2.zig:885:18: error: container 'crypto.test' has no member called 'assertEqualHash'
        try htest.assertEqualHash(Blake2b512, h1, block[0..]);
                 ^
./lib/std/crypto/md5.zig:240:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Md5, "d41d8cd98f00b204e9800998ecf8427e", "");
             ^
./lib/std/crypto/md5.zig:254:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("d41d8cd98f00b204e9800998ecf8427e", out[0..]);
             ^
./lib/std/crypto/sha1.zig:276:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Sha1, "da39a3ee5e6b4b0d3255bfef95601890afd80709", "");
             ^
./lib/std/crypto/sha1.zig:286:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("da39a3ee5e6b4b0d3255bfef95601890afd80709", out[0..]);
             ^
./lib/std/crypto/sha2.zig:296:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Sha224, "d14a028c2a3a2bc9476102bb288234c415a2b01f828ea62ac5b3e42f", "");
             ^
./lib/std/crypto/sha2.zig:306:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("d14a028c2a3a2bc9476102bb288234c415a2b01f828ea62ac5b3e42f", out[0..]);
             ^
./lib/std/crypto/sha2.zig:322:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Sha256, "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855", "");
             ^
./lib/std/crypto/sha2.zig:332:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855", out[0..]);
             ^
./lib/std/crypto/sha2.zig:687:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Sha384, h1, "");
             ^
./lib/std/crypto/sha2.zig:702:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/sha2.zig:721:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Sha512, h1, "");
             ^
./lib/std/crypto/sha2.zig:736:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/sha3.zig:180:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Sha3_224, "6b4e03423667dbb73b6e15454f0eb1abd4597f9a1b078e3f5b5a6bc7", "");
             ^
./lib/std/crypto/sha3.zig:190:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("6b4e03423667dbb73b6e15454f0eb1abd4597f9a1b078e3f5b5a6bc7", out[0..]);
             ^
./lib/std/crypto/sha3.zig:206:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Sha3_256, "a7ffc6f8bf1ed76651c14756a061d662f580ff4de43b49fa82d80a4b80f8434a", "");
             ^
./lib/std/crypto/sha3.zig:216:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("a7ffc6f8bf1ed76651c14756a061d662f580ff4de43b49fa82d80a4b80f8434a", out[0..]);
             ^
./lib/std/crypto/sha3.zig:242:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Sha3_384, h1, "");
             ^
./lib/std/crypto/sha3.zig:255:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/sha3.zig:273:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Sha3_512, h1, "");
             ^
./lib/std/crypto/sha3.zig:286:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(h1, out[0..]);
             ^
./lib/std/crypto/sha3.zig:312:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Keccak_256, "c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470", "");
             ^
./lib/std/crypto/sha3.zig:318:14: error: container 'crypto.test' has no member called 'assertEqualHash'
    try htest.assertEqualHash(Keccak_512, "0eab42de4c3ceb9235fc91acffe746b29c29a8c366b7c60e4e67c466f36a4304c00fa9caf9d87976ba469bcbe06713b435f091ef2769fb160cdab33d3670680e", "");
             ^
./lib/std/crypto/gimli.zig:286:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("1C9A03DC6A5DDC5444CFC6F4B154CFF5CF081633B2CEA4D7D0AE7CCFED5AAA44", &md);
             ^
./lib/std/crypto/gimli.zig:294:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("404C130AF1B9023A7908200919F690FFBB756D5176E056FFDE320016A37C7282", &md);
             ^
./lib/std/crypto/gimli.zig:302:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("A8F4FA28708BDA7EFB4C1914CA4AFA9E475B82D588D36504F87DBB0ED9AB3C4B", &md);
             ^
./lib/std/crypto/gimli.zig:459:18: error: container 'crypto.test' has no member called 'assertEqual'
        try htest.assertEqual("", &ct);
                 ^
./lib/std/crypto/hmac.zig:82:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("74e6f7298a9c2d168935f58c001bad88", out[0..]);
             ^
./lib/std/crypto/hmac.zig:91:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("fbdb1d1b18aa6c08324b7d64b71fb76370690e1d", out[0..]);
             ^
./lib/std/crypto/hmac.zig:100:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("b613679a0814d9ec772f95d778c35fc5ff1697c493715653c6c712144292c5ad", out[0..]);
             ^
./lib/std/crypto/25519/x25519.zig:176:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("90e7595fc89e52fdfddce9c6a43d74dbf6047025ee0462d2d172e8b6a2841d6e", &mont_kp.secret_key);
             ^
./lib/std/crypto/25519/edwards25519.zig:538:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("0691eee3cf70a0056df6bfa03120635636581b5c4ea571dfc680f78c7e0b4137", p.toBytes()[0..]);
             ^
./lib/std/crypto/25519/edwards25519.zig:548:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("31558a26887f23fb8218f143e69d5f0af2e7831130bd5b432ef23883b895839a", p.toBytes()[0..]);
             ^
./lib/std/crypto/hkdf.zig:62:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("077709362c2e32df0ddc3f0dc47bba6390b6c73bb50f9c3122ec844ad7c2b3e5", &prk);
             ^
./lib/std/crypto/ghash.zig:325:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("889295fa746e8b174bf4ec80a65dea41", &out);
             ^
./lib/std/crypto/pbkdf2.zig:165:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(expected, dk[0..]);
             ^
./lib/std/crypto/pbkdf2.zig:180:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(expected, dk[0..]);
             ^
./lib/std/crypto/pbkdf2.zig:195:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(expected, dk[0..]);
             ^
./lib/std/crypto/pbkdf2.zig:230:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(expected, dk[0..]);
             ^
./lib/std/crypto/pbkdf2.zig:245:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual(expected, dk[0..]);
             ^
./lib/std/crypto/salsa20.zig:559:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("30ff9933aa6534ff5207142593cd1fca4b23bdd8", c[0..]);
             ^
./lib/std/crypto/salsa20.zig:626:14: error: container 'crypto.test' has no member called 'assertEqual'
    try htest.assertEqual("b05760e217288ba079caa2fd57fd3701784974ffcfda20fe523b89211ad8af065a6eb37cdb29d51aca5bd75dafdd21d18b044c54bb7c526cf576c94ee8900f911ceab0147e82b667a28c52d58ceb29554ff45471224d37b03256b01c119b89ff6d36855de8138d103386dbc9d971f52261", &ciphertext);
             ^

```

--- a/lib/std/os/linux/io_uring.zig
+++ b/lib/std/os/linux/io_uring.zig
@@ -1840,42 +1840,73 @@ test "timeout_remove" {
     }, cqe_timeout_remove);
 }
 
-test "timeout_link_chain1" {
+test "accept/connect/recv/link_timeout" {
     if (builtin.os.tag != .linux) return error.SkipZigTest;
 
-    var ring = IO_Uring.init(8, 0) catch |err| switch (err) {
+    var ring = IO_Uring.init(16, 0) catch |err| switch (err) {
         error.SystemOutdated => return error.SkipZigTest,
         error.PermissionDenied => return error.SkipZigTest,
         else => return err,
     };
     defer ring.deinit();
 
-    var fds = try os.pipe();
-    defer {
-        os.close(fds[0]);
-        os.close(fds[1]);
+    const address = try net.Address.parseIp4("127.0.0.1", 3131);
+    const kernel_backlog = 1;
+    const server = try os.socket(address.any.family, os.SOCK.STREAM | os.SOCK.CLOEXEC, 0);
+    defer os.close(server);
+    try os.setsockopt(server, os.SOL.SOCKET, os.SO.REUSEADDR, &mem.toBytes(@as(c_int, 1)));
+    try os.bind(server, &address.any, address.getOsSockLen());
+    try os.listen(server, kernel_backlog);
+
+    var buffer_recv = [_]u8{ 0, 1, 0, 1, 0 };
+
+    var accept_addr: os.sockaddr = undefined;
+    var accept_addr_len: os.socklen_t = @sizeOf(@TypeOf(accept_addr));
+    _ = try ring.accept(0xaaaaaaaa, server, &accept_addr, &accept_addr_len, 0);
+    try testing.expectEqual(@as(u32, 1), try ring.submit());
+
+    const client = try os.socket(address.any.family, os.SOCK.STREAM | os.SOCK.CLOEXEC, 0);
+    defer os.close(client);
+    _ = try ring.connect(0xcccccccc, client, &address.any, address.getOsSockLen());
+    try testing.expectEqual(@as(u32, 1), try ring.submit());
+
+    var cqe_accept = try ring.copy_cqe();
+    if (cqe_accept.err() == .INVAL) return error.SkipZigTest;
+    var cqe_connect = try ring.copy_cqe();
+    if (cqe_connect.err() == .INVAL) return error.SkipZigTest;
+
+    // The accept/connect CQEs may arrive in any order, the connect CQE will sometimes come first:
+    if (cqe_accept.user_data == 0xcccccccc and cqe_connect.user_data == 0xaaaaaaaa) {
+        const a = cqe_accept;
+        const b = cqe_connect;
+        cqe_accept = b;
+        cqe_connect = a;
     }
 
-    var buffer = [_]u8{0} ** 128;
-    const iovecs = [_]os.iovec{os.iovec{ .iov_base = &buffer, .iov_len = buffer.len }};
-    const sqe_readv = try ring.readv(0x11111111, fds[0], &iovecs, 0);
-    sqe_readv.flags |= linux.IOSQE_IO_LINK;
+    try testing.expectEqual(@as(u64, 0xaaaaaaaa), cqe_accept.user_data);
+    if (cqe_accept.res <= 0) std.debug.print("\ncqe_accept.res={}\n", .{cqe_accept.res});
+    try testing.expect(cqe_accept.res > 0);
+    try testing.expectEqual(@as(u32, 0), cqe_accept.flags);
+    try testing.expectEqual(linux.io_uring_cqe{
+        .user_data = 0xcccccccc,
+        .res = 0,
+        .flags = 0,
+    }, cqe_connect);
+
+    const sqe_recv = try ring.recv(0xffffffff, cqe_accept.res, buffer_recv[0..], 0);
+    sqe_recv.flags |= linux.IOSQE_IO_LINK;
 
     const ts = os.linux.kernel_timespec{ .tv_sec = 0, .tv_nsec = 1000000 };
-    const seq_link_timeout = try ring.link_timeout(0x22222222, &ts, 0);
-    seq_link_timeout.flags |= linux.IOSQE_IO_LINK;
-
-    _ = try ring.nop(0x33333333);
+    _ = try ring.link_timeout(0x22222222, &ts, 0);
 
     const nr_wait = try ring.submit();
-    try testing.expectEqual(@as(u32, 3), nr_wait);
+    try testing.expectEqual(@as(u32, 2), nr_wait);
 
     var i: usize = 0;
     while (i < nr_wait) : (i += 1) {
         const cqe = try ring.copy_cqe();
         switch (cqe.user_data) {
-            // poll cancel really should return -ECANCEL...
-            0x11111111 => {
+            0xffffffff => {
                 if (cqe.res != -@as(i32, @enumToInt(linux.E.INTR)) and
                     cqe.res != -@as(i32, @enumToInt(linux.E.CANCELED)))
                 {
@@ -1884,20 +1915,9 @@ test "timeout_link_chain1" {
                 }
             },
             0x22222222 => {
-                // FASTPOLL kernels can cancel successfully
                 if (cqe.res != -@as(i32, @enumToInt(linux.E.ALREADY)) and
                     cqe.res != -@as(i32, @enumToInt(linux.E.TIME)))
                 {
-                    if (cqe.res == -@as(i32, @enumToInt(linux.E.BADF))) {
-                        // https://github.com/ziglang/zig/issues/10247
-                        return error.SkipZigTest;
-                    }
-                    std.debug.print("Req 0x{x} got {d}\n", .{ cqe.user_data, cqe.res });
-                    try testing.expect(false);
-                }
-            },
-            0x33333333 => {
-                if (cqe.res != -@as(i32, @enumToInt(linux.E.CANCELED))) {
                     std.debug.print("Req 0x{x} got {d}\n", .{ cqe.user_data, cqe.res });
                     try testing.expect(false);
                 }

